### PR TITLE
updating treadmill analysis

### PR DESCRIPTION
**Pull request by kharrison89**
Run: creating analaysis based on walk. accomodates level/uphill/downhill. kinematics analysis can be toggled on/off. 

Walk: adding export of bad files to be followed up on.

### CODE REVIEW

File: Python/Performance_Run_ExtractKinematicKineticVars.py [hunk 1]
```diff
@@ -0,0 +1,824 @@
...
+
+        # Extract data frequency
+        freq = pd.read_csv(fPath+fName,sep='\t',usecols=[0], nrows=1, skiprows=[0,1], header = 0)
+        freq = freq.values.tolist()
+        freq = freq[0][0]
+        
+        # Extract subject mass
+        mass = pd.read_csv(fPath+fName,sep='\t',usecols=[0], nrows=1, skiprows= 6, header = None)
+        mass = mass.values.tolist()
+        mass = mass[0][0]
```
Comment (dfeeney31) [inline Python/Performance_Run_ExtractKinematicKineticVars.py:650]: hey kate, noticed some files have mass missing. here is a way to maybe fix it.

File: Python/Performance_Run_ExtractKinematicKineticVars.py [hunk 2]
```diff
@@ -0,0 +1,740 @@
...
+            tmpCond = 'Downhill'
+            tmpSlope = -10
+            # Set the angle of the treadmill
+            ang = 10
+            
+            
+        elif fName.count('UH'):
+            speed = 1.2
+            tmpCond = 'Uphill'
+            tmpSlope = 10
```
Comment (EricHonertBOA) [inline Python/Performance_Run_ExtractKinematicKineticVars.py:526]: Changed to only have the conversion be within the COM power function

File: Python/Performance_Run_ExtractKinematicKineticVars.py [hunk 3]
```diff
@@ -0,0 +1,740 @@
...
+    """
+    # Compute the COM power using the individual limbs method                
+    
+    # Debugging tool: Showing the time-continuous COM power
+    show_COMpower = 0
+
+    # First compute the approximate body weight: will need to rotate the
+    # ground reaction forces into the inertial coordinate system
+    slope = slope*np.pi/180
+    BM = np.nanmean(GRF[:,1]*np.sin(slope*np.pi/180)+GRF[:,2]*np.cos(slope))/9.81
```
Comment (EricHonertBOA) [inline Python/Performance_Run_ExtractKinematicKineticVars.py:333]: Updating to let body mass be defined by GRFs within the code

File: Python/Performance_Run_ExtractKinematicKineticVars.py [hunk 4]
```diff
@@ -0,0 +1,740 @@
...
+        else:
+            dat.Left_GRF_Y = -1 * dat.Left_GRF_Y
+        
+        # Extract subject mass
+        mass = pd.read_csv(fPath+fName,sep='\t',usecols=[0], nrows=1, skiprows= 6, header = None)
+        mass = mass.values.tolist()
+        mass = mass[0][0]
+        if type(mass) != float:
+            print('Mass missing from file! Computing mass from GRFs')
+            mass = np.nanmean(dat.Left_GRF_Z)/9.81
```
Comment (EricHonertBOA) [inline Python/Performance_Run_ExtractKinematicKineticVars.py:568]: if the mass is not within the file, compute it from the vertical GRFs: note, this assume vertical GRFs

File: Python/Performance_Run_ExtractKinematicKineticVars.py [hunk 5]
```diff
@@ -0,0 +1,740 @@
...
+                GS.append(jj)
+        GS = np.array(GS)
+        
+        # Compute COM work
+        [tmpCW_pos,tmpCW_neg,debug_COMPW] = COMPower_Work_run(LGRF,speed,ang,trimmedLandings,trimmedTakeoffs,GS,freq)
+        COMWork_pos.extend(tmpCW_pos)
+        COMWork_neg.extend(tmpCW_neg)
+        
+        #______________________________________________________________
+        # Debugging: Creation of dialog box for looking where foot contact are accurate
```
Comment (EricHonertBOA) [inline Python/Performance_Run_ExtractKinematicKineticVars.py:620]: added debugging plots along with dialog box.

File: Python/Performance_Run_ExtractKinematicKineticVars.py [hunk 6]
```diff
@@ -0,0 +1,740 @@
...
+            plt.xlabel('% Step')
+            
+            answer = messagebox.askyesno("Question","Is data clean?")
+            plt.close()
+            if answer == False:
+                print('Adding file to bad file list')
+                badFileList.append(fName)
+            
+        if answer == True:
+            if kinematics == 1:# Distal foot power computations
```
Comment (EricHonertBOA) [inline Python/Performance_Run_ExtractKinematicKineticVars.py:646]: note: we may have to update the code in the future to accommodate kinematics. the current form of debugging and computations are more geared towards just the GRF based metrics

**Review by EricHonertBOA — APPROVED** [review]

**Pull request merged**